modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
Yes, I agree. Therefore, it's possible to:

* dynamically add/remove the "unorderedness" to any node with 0 children; or
* statically define a node type (regular or mixin) as being "unordered" by extending one of the unordered mixins (and using that as a primary type for nodes).

Fantastic.